### PR TITLE
try pypy3.8 + meson

### PR DESCRIPTION
**Pull request by mattip**

### CODE REVIEW

File: .github/workflows/linux_meson.yml [hunk 1]
```diff
@@ -26,11 +26,11 @@ jobs:
   test_meson:
     name: Meson build
     # If using act to run CI locally the github object does not exist and the usual skipping should not be enforced
-    if: "github.repository != '' || github.repository == 'scipy/scipy' && !contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.head_commit.message, '[skip ci]') && !contains(github.event.head_commit.message, '[skip github]') && !contains(github.ref, 'maintenance/') && !contains(github.base_ref, 'maintenance/')"
+    # if: "github.repository != '' || github.repository == 'scipy/scipy' && !contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.head_commit.message, '[skip ci]') && !contains(github.event.head_commit.message, '[skip github]') && !contains(github.ref, 'maintenance/') && !contains(github.base_ref, 'maintenance/')"
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.9]
+        python-version: [3.9, pypy-3.8-v7.3.8]
```
Comment (mattip) [inline .github/workflows/linux_meson.yml]:
```suggestion
        python-version: [3.9, pypy-3.8-nightly]
```

**Comment by eli-schwartz** [comment]
> ERROR: Clock skew detected. File D:\a\_temp\44e9eef4-181f-4198-bccc-ecef321c4565\pypy-c-jit-105642-c5275712f35d-win64\python.exe has a time stamp 10137.8583s in the future.

This looks like a build flake and can maybe just be restarted.

**Comment by mattip** [comment]
The clock skew is because meson is checking timestamps of system files, the timestamps on windows are local time, and the machine building the zip is in a timezone ahead of the machine using the zip. This seems to me to be something meson should be taught to ignore on windows. Otherwise a user downloading a new zip file from New Zealand and running meson in San Francisco will have to wait 20 hours to use the meson build.

**Comment by eli-schwartz** [comment]
https://github.com/mesonbuild/meson/blob/f9200ca8f7429d614dcb8c4144269461c6a7d9d9/mesonbuild/backend/backends.py#L1195-L1209

Ignoring this is unlikely to work as it will just cause ninja to endlessly try reconfiguring.

Ideally timestamps on files would be synced across the globe by using UTC with a real timezone of course...

**Comment by mattip** [comment]
[CPython3.9](https://github.com/mattip/scipy/runs/6641771854?check_suite_focus=true#step:3:9) testing with [scipy HEAD](https://github.com/mattip/scipy/runs/6641771854?check_suite_focus=true#step:2:556) is failing [`test_all_modules_are_expected_2` and `test_modules_importable`](https://github.com/mattip/scipy/runs/6641771854?check_suite_focus=true#step:12:2343).

**Comment by mattip** [comment]
> [CPython3.9](https://github.com/mattip/scipy/runs/6641771854?check_suite_focus=true#step:3:9) testing with [scipy HEAD](https://github.com/mattip/scipy/runs/6641771854?check_suite_focus=true#step:2:556) is failing [test_all_modules_are_expected_2 and test_modules_importable](https://github.com/mattip/scipy/runs/6641771854?check_suite_focus=true#step:12:2343).

rebased off scipy/scipy main to pick up fixes

**Comment by mattip** [comment]
On windows, `pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy==1.23.0.dev0+1087.g0eaa40db3` builds successfully. This was uploaded April 28.

**Comment by mattip** [comment]
There is a [fix to cython](https://github.com/cython/cython/pull/4764) for the [`ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__`](https://github.com/mattip/scipy/runs/6645761326?check_suite_focus=true#step:12:912) errors. It has not made it into a release yet. I am using the fixed version here for scipy, but NumPy is built without the fix.

**Pull request closed without merging**